ztp/siteconfig-generator/siteConfig/siteConfigHelper.go

```diff
@@ -0,0 +1,129 @@
+package siteConfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	mcfgctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+
+	"github.com/lack/yamltrim"
+	yaml "gopkg.in/yaml.v3"
+)
+
+const mcRoleKey = "machineconfiguration.openshift.io/role"
+const mcName = "predefined-extra-manifests"
+
+func isExcluded(s []string, str string) bool {
+	if s == nil {
+		return false
+	}
+	for _, v := range s {
+		if str == v {
+			return true
+		}
+	}
+	return false
+}
+
+// merge the spec fields of all MC manifests except the ones that are in the excluded list
+func MergeManifests(dataMap map[string]interface{}, excludes []string) (map[string]interface{}, error) {
+	// key is role, value is a list of MCs
+	configs := make(map[string][]*machineconfigv1.MachineConfig)
+
+	for k, v := range dataMap {
+		if isExcluded(excludes, k) {
+			continue
+		}
+		err := addMachineConfig(v, configs)
+		if err != nil {
+			return dataMap, err
+		}
+		// remove the individual file entries
+		delete(dataMap, k)
+	}
+
+	for k, v := range configs {
```

Comment:
In this case `k`, `v` may be better as `roleName`, `machineConfigs`?

Reply:
Yes.